src/Microsoft.SqlTools.ResourceProvider.DefaultImpl/AzureResourceManager.cs

```diff
@@ -55,12 +55,19 @@ public Task<IAzureResourceManagementSession> CreateSessionAsync(IAzureUserAccoun
             CommonUtil.CheckForNull(subscriptionContext, "subscriptionContext");
             try
             {
+                string armEndpoint = subscriptionContext.UserAccount.UnderlyingAccount.Properties.ProviderSettings?.Settings?.ArmResource?.Endpoint;
+                Uri armUri = null;
+                if (armEndpoint != null)
+                {
+                    armUri = new Uri(armEndpoint);
```

Comment:
Consider for these having it fall back to using the credentials if parsing the URI fails

Reply:
I think I'm doing that here: https://github.com/microsoft/sqltoolsservice/pull/920/files/28765aa33ebcf5b8f59e9ae3806e37c4851120a2#diff-68c0bc2a8a7ae81b805166ef253c3696R65

Reply:
I'm saying if it throws an exception while parsing (since that's what'll happen if armEndpoint isn't a valid URI)

Reply:
Ah I see, that'd be an extremely weird case but I'll plan for it too.